packages/cardano/src/wallet/lib/cardano-wallet.ts

```diff
@@ -27,7 +28,10 @@ import { Wallet } from '@src/index';
 import { HexBlob } from '@cardano-sdk/util';
 import { WsProvider } from '@cardano-sdk/cardano-services-client';
 
-export const bip32Ed25519 = new Crypto.SodiumBip32Ed25519();
+let bip32Ed25519: Crypto.SodiumBip32Ed25519;
+
+export const getBip32Ed25519 = async (): Promise<Crypto.SodiumBip32Ed25519> =>
+  bip32Ed25519 || (bip32Ed25519 = await Crypto.SodiumBip32Ed25519.create());
```

Comment:
Minor improvement: `Promise.all([getBip32Ed25519(), getBip32Ed25519()])` would currently initialize 2 objects. If we remove `await` then it'll always be 1.
```suggestion
let bip32Ed25519: Promise<Crypto.SodiumBip32Ed25519>;

export const getBip32Ed25519 = (): Promise<Crypto.SodiumBip32Ed25519> =>
  bip32Ed25519 || (bip32Ed25519 = Crypto.SodiumBip32Ed25519.create());
```

Reply:
Fixed, thank you! 😄

Reply:
I had to add `let bip32Ed25519: Promise<Crypto.SodiumBip32Ed25519> | undefined;` or otherwise Sonar was complaining.